### ISSUE TITLE
installing packages with bun causes runtime error in dev-build (expo-sqlite)

### ISSUE THREAD

**Issue by RITE-Soft** [expo/expo#27420]
### Summary

When installing packages with `bun`, the dev-build (`bun run:android`) will throw a runtime error in `expo-sqlite` on android virtual devices:
```
 TypeError: Cannot read property 'prototype' of undefined, js engine: hermes
```

The error occurs in `SQLiteDatabase.js 291:58` in expo-sqlite

**_However, when installing the same `package.json` with _npm_ or _yarn_, this error does not occur in the resulting dev-build._**

Error:
<img src="https://i.imgur.com/AwI72T9.png" data-canonical-src="https://i.imgur.com/AwI72T9.png" width="200" />

_____________________________________________________
### Managed or bare workflow?

managed

### What platform(s) does this occur on?

Android

### Package versions

    "babel-plugin-inline-import": "^3.0.0",
    "drizzle-orm": "^0.29.4",
    "expo": "~50.0.8",
    "expo-dev-client": "^3.3.9",
    "expo-sqlite": "^13.2.2",
    "expo-status-bar": "~1.11.1",
    "react": "18.2.0",
    "react-native": "0.73.4"

### Environment

expo-env-info 1.2.0 environment info:
    System:
      OS: Linux 5.15 Ubuntu 20.04.6 LTS (Focal Fossa)
      Shell: 5.0.17 - /bin/bash
    Binaries:
      Node: 18.19.1 - ~/.nvm/versions/node/v18.19.1/bin/node
      Yarn: 1.22.21 - /mnt/c/Program Files/nodejs/yarn
      npm: 10.2.4 - ~/.nvm/versions/node/v18.19.1/bin/npm
    npmPackages:
      expo: ~50.0.8 => 50.0.8
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.73.4 => 0.73.4 
    Expo Workflow: bare

### Reproducible demo

~~https://github.com/0tii/expo-template-typescript-sqlite-orm-nativewind-jest~~

Even more minimal example (bare typescript template with sqlite & drizzle):

https://github.com/0tii/bun-breaks-sqlite-minimal-example

**Comment by RITE-Soft** [comment]
Migrating the `package-lock.json`, that has been tested to work with `npm run android`, to` bun.lockb` with `bun pm migrate` and doing a clean install with `bun install --frozen-lockfile`, the error will still occur.

Doing a `npm ci` with that same lock file will not cause this error and result in a working build.

Doing a bun `run:android` after installing packages with either yarn or npm will result in a working build.

After every attempt, `node_modules/` and `android/` are deleted.

**Comment by byCedric** [comment]
Hi @RITE-Soft! Thanks for reporting this issue. This may actually be related to my Bun issue report from a while ago: https://github.com/oven-sh/bun/issues/8406.

Unfortunately, I can't confirm this as I can't reproduce this issue. If it's not oven-sh/bun#8406, it might be related to the native side of `expo-sqlite/next`.

Can you check if the expected library versions of `drizzle-orm` and `expo-sqlite`, _and their dependencies_, are installed correctly with Bun?

**Comment by Kudo** [comment]
cannot repro on the https://github.com/0tii/bun-breaks-sqlite-minimal-example as well.
my bun version is 1.0.23
my steps:

```sh
$ git clone https://github.com/0tii/bun-breaks-sqlite-minimal-example
$ cd bun-breaks-sqlite-minimal-example
$ bun i --frozen-lockfile
$ npx expo run:android
```

please let me know if i missed something.

**Comment by RITE-Soft** [comment]
@byCedric @Kudo I am using bun 1.0.28 and 1.0.29, so I will downgrade to 1.0.23 tomorrow and try again. For me it's infinitely reproducible on my WSL install, but i will also give it a try on my ubuntu machine to verify that the issue might be restricted to my WSL distro

**Comment by RITE-Soft** [comment]
@Kudo I just uninstalled and downgraded bun to `1.0.23` and indeed the error is gone. I upgraded back to `1.0.29` and it did not reappear again. I am majorly confused. I then investigated if it might have been, because I installed bun from `npm` initially and could not reproduce it again.

It looks to me like the bun reinstall magically fixed this issue, which might have arisen from a broken initial bun install, because otherwise I can not explain this, given I have tested this breaking legitimately 10 times between yarn and bun before creating this issue 😅
